cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java

```diff
@@ -653,8 +661,17 @@ private void assertConsistentTypes(Class<? extends Mapper> firstMapperClass,
       // A bit hacky for stream.
       if (provider instanceof StreamInputFormatProvider) {
         // pass in mapperInput.getMapper() instead of mapperClass, because mapperClass defaults to the Identity Mapper
-        setDecoderForStream((StreamInputFormatProvider) provider, job, inputFormatConfiguration,
-                            mapperInput.getMapper());
+        StreamInputFormatProvider inputFormatProvider = (StreamInputFormatProvider) provider;
+        setDecoderForStream(inputFormatProvider, job, inputFormatConfiguration, mapperInput.getMapper());
+        // Check if the MR job has read access to the stream, if not fail right away. Note that this is being done
```

Comment:
I checked with @poornachandra and @chtyim and they mentioned that we want to capture lineage/usage even if the read didn't actually succeed.